src/App/Fossa/VPS/AOSPNotice.hs

```diff
@@ -0,0 +1,47 @@
+module App.Fossa.VPS.AOSPNotice
+  ( aospNoticeMain,
+    WriteEnabled(..)
+  ) where
+
+import Control.Effect.Lift (Lift)
+import Control.Carrier.Diagnostics
+import Effect.Exec
+import System.Exit (exitFailure)
+
+import App.Fossa.EmbeddedBinary
+import App.Fossa.VPS.Scan.RunWiggins
+import App.Fossa.VPS.Types
+import App.Types (BaseDir (..))
+import Data.Flag (Flag, fromFlag)
+import Effect.Logger ( Severity, withLogger, logInfo )
+import Data.Text (Text)
+import Data.Text.Prettyprint.Doc
+
+-- | WriteEnabled bool flag
+data WriteEnabled = WriteEnabled
+
+aospNoticeMain :: BaseDir -> Severity -> FilterExpressions -> Flag WriteEnabled ->  IO ()
+aospNoticeMain basedir logSeverity fileFilters writeEnabled = do
+  result <- runDiagnostics $ withWigginsBinary $ aospNoticeGenerate basedir logSeverity writeEnabled fileFilters
```

Comment:
This should probably be wrapped with a `withLogger logSeverity $ ...` instead of passing the logging responsibility downwards.

Reply:
Something like 
```
withLogger logSeverity $ do
  result <- runDiagnostics $ undefined
  -- handle diagnostics
```

Reply:
We use `Severity` to determine whether we should pass a `-debug` to Wiggins. Maybe I'm not visualizing what you mean properly, but I think we wouldn't be able to do so with this change.

Reply:
You can still pass the `logSeverity` value down (it's just an empty sum type), but you would actually setup the logging effect here.  Lmk if you want to chat more, I might be assuming some knowledge of the effects that isn't clear here.